From 186bcfe4c81f7bb48d0495fb4fef5cf855f12adc Mon Sep 17 00:00:00 2001
From: Vincent Bernat <vincent@bernat.im>
Date: Sat, 28 Apr 2018 17:31:42 +0200
Subject: [PATCH] Ensure NoNextHop stays a singleton

In 58e809a8549f9f0f1e5372d4c800c090275d9b9c, a deep copy is done. This
makes a second "NoNextHop" instance while we believe it should be a
singleton. Implement `__copy__` and `__deepcopy__` to ensure we keep a
singleton.
---
 lib/exabgp/protocol/ip/__init__.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/lib/exabgp/protocol/ip/__init__.py b/lib/exabgp/protocol/ip/__init__.py
index bd690332367c..2ce27efeeee3 100644
--- a/lib/exabgp/protocol/ip/__init__.py
+++ b/lib/exabgp/protocol/ip/__init__.py
@@ -226,6 +226,12 @@ class _NoNextHop (object):
 	def __str__ (self):
 		return 'no-nexthop'
 
+	def __deepcopy__(self, _):
+		return self
+
+	def __copy__(self, _):
+		return self
+
 
 NoNextHop = _NoNextHop()
 
-- 
2.17.0

